### ISSUE TITLE
Error on non-zero exit statuses for pipeline elements

### ISSUE THREAD

**Issue by Hofer-Julian** [nushell/nushell#13817]
### Related problem

Since https://github.com/nushell/nushell/pull/13515 non-zero exit status of a command on a trailing pipeline position errors. The same isn't true for pipeline elements.

So `a)` will error, while `b)` will not.
Assuming that `b)` doesn't collect its input.
```nushell
# a)
foo | may-fail
# b)
foo | may-fail | bar
```

### Describe the solution you'd like

According to @ianmanske, changing that [isn't possible with the current code structure](https://github.com/nushell/nushell/pull/13515#issuecomment-2318037638)

I still wanted to open an issue, since this might change at some point. If it does, I would appreciate it if nushell would error for **both** scenarios `a)` and `b)`

**Comment by kik4444** [comment]
If this is included, will there or *should* there be any options to enable/disable this behavior like Bash's `set -e` or do we want to make it always enabled? I'd like Nushell not to exhibit totally different behavior on the same code depending on external factors (i.e. the `config.nu`). I believe this has already been mentioned in #13515. I'm sure people wouldn't like to make their `config.nu` a hidden dependency of some script.

Personally, I'd be in favor of making this enabled all the time, provided that #13868 is fixed and that wrapping the external command in a `try {}` is enough to avoid prematurely exiting your code.

For example `"1\n2" | grep "a" | detect columns` should raise an error, while `"1\n2" | try { grep "a" } | detect columns` should return `nothing`

**Comment by IanManske** [comment]
Yeah, I agree. If this is implemented, it will be always enabled and without the ability for configuration. But since it would not be configurable, there would need to be some other way to suppress errors for certain elements/commands. So, `do` or `try` would need to have this functionality like you pointed out.

**Comment by GrabbenD** [comment]
Pipefail is really useful but can be problematic in automated bash scripts, for instance when unmounting paths which are already unmounted.

Hence the ability to enable/disable it globally and to be able to toggle it for individual expressions would go a really far way!

**Comment by miniluz** [comment]
A workaround, if you only need to run one external command and you don't wanna use complete because you want the output to be printed gradually to stdout instead of all at once at the end:

```nu
> nu -c 'print "hey" ; sleep 1sec ; print "hey" ; exit 1' | tee { print } | tee { save output } | complete
hey
hey
╭───────────┬─────╮
│ stdout    │ hey │
│           │ hey │
│           │     │
│ stderr    │     │
│ exit_code │ 1   │
╰───────────┴─────╯
```

(If you try doing `nu -c 'print "hey" ; sleep 1sec ; print "hey" ; exit 1' | tee { save output }` and access `$env.LAST_EXIT_CODE` it is `0`)
